background_isolate_channels/lib/simple_database.dart

```diff
@@ -0,0 +1,236 @@
+    if (!file.existsSync()) {
+      file.createSync();
+    }
+    RandomAccessFile writer = await file.open(mode: FileMode.append);
+    List<int> bytes = utf8.encode(value);
+    if (bytes.length > entrySize) {
+      bytes = bytes.sublist(0, entrySize);
+    } else if (bytes.length < entrySize) {
+      List<int> newBytes = List.filled(entrySize, 0);
+      for (int i = 0; i < bytes.length; ++i) {
+        newBytes[i] = bytes[i];
+      }
+      bytes = newBytes;
+    }
+    await writer.writeFrom(bytes);
+    await writer.close();
+    _sendPort.send(_Command(_Codes.ack, <dynamic>[null]));
+  }
+
+  /// Perform the find entry operation.
+  Future<void> _doFind(String query) async {
+    if (isDebug) {
+      print('Performing find: $query');
+    }
+    File file = File(_path!);
+    if (file.existsSync()) {
+      RandomAccessFile reader = file.openSync();
+      List<int> buffer = List.filled(entrySize, 0);
+      while (reader.readIntoSync(buffer) == entrySize) {
+        List<int> foo = buffer.takeWhile((value) => value != 0).toList();
+        String string = utf8.decode(foo);
+        if (string.contains(query)) {
+          _sendPort.send(_Command(_Codes.result, <dynamic>[string]));
+        }
+      }
+      reader.closeSync();
+    }
+    _sendPort.send(_Command(_Codes.done, <dynamic>[null]));
+  }
+
+  /// Handle the [command] received from the [ReceivePort].
+  Future<void> _onCommand(_Command command) async {
+    if (command.code == _Codes.init) {
+      _path = command.args[0];
+      // ----------------------------------------------------------------------
+      // The [RootIsolateToken] is required for
+      // [BackgroundIsolateBinaryMessenger.ensureInitialized] and must be
+      // obtained on the root isolate and passed into the background isolate via
+      // a [SendPort].
+      // ----------------------------------------------------------------------
+      RootIsolateToken rootIsolateToken = command.args[1];
+      // ----------------------------------------------------------------------
+      // [BackgroundIsolateBinaryMessenger.ensureInitialized] must be called
+      // before using any plugins. This sets up the [BinaryMessenger] that the
+      // Platform Channels will communicate with on the background isolate.
+      // ----------------------------------------------------------------------
+      BackgroundIsolateBinaryMessenger.ensureInitialized(rootIsolateToken);
+      _sharedPreferences = await SharedPreferences.getInstance();
+    } else if (command.code == _Codes.add) {
+      await _doAddEntry(command.args[0]);
+    } else if (command.code == _Codes.query) {
+      _doFind(command.args[0]);
+    }
+  }
+}
+
+/// A SimpleDatabase that stores entries of strings to disk where they can be
+/// queried.
+///
+/// All the disk operations and queries are executed in a background isolate
+/// operating. This class just sends and receives messages to the isolate.
+class SimpleDatabase {
```

Comment:
I would sort this before the private class for better reading flow.

Reply:
done